test/cat.jl

```diff
@@ -118,17 +109,8 @@ module TestCat
     dtc = DataTable(a = NullableArray([2, 3, 4]))
     dtd = DataTable(Any[2:4], [:a])
     dtab = vcat(dta, dtb)
-    dtac = vcat(dta, dtc)
```

Comment:
Actually, since support in Base won't come before Julia 1.0, we should use a custom promotion approach for now, just like we did before this PR. Instead of calling `vcat`, we should first compute the type of the column, then allocate it, and `copy!` the successive inputs into it. The computation of the type can be done using a strategy similar to `similar_nullable`, something like:
```julia
@generated function promote_col_type(cols::AbstractVector...)
    elty = Base.promote_eltype(cols...)
    if elty <: Nullable
        elty = eltype(elty)
    end
    if elty <: CategoricalValue
        elty = elty.parameters[1]
    end
    if any(col -> eltype(col) <: Nullable, cols)
        if any(col -> col <: Union{AbstractCategoricalArray, AbstractNullableCategoricalArray}, cols)
            return :(NullableCategoricalVector{$elty})
        else
            return :(NullableVector{$elty})
        end
    else
        if any(col -> col <: Union{AbstractCategoricalArray, AbstractNullableCategoricalArray}, cols)
            return :(CategoricalVector{$elty})
        else
            return :(Vector{$elty})
        end
    end
end
```